test/unit/networking.test.js

```diff
@@ -5,16 +5,96 @@ import sinon from 'sinon';
+
+      const origin = networking.getStandardOrigin();
+
+      assert.equal(origin, "https://example.pubnubapi.com");    
+    });
+
+    it('should generate sharded domains based on default origin', () => {
+      const config = new Config({ setup: { ssl: true } })
+      const networking = new Networking({})
+      networking.init(config);
+
+      const origin = networking.getStandardOrigin();
+
+      const regex = /^https:\/\/ps([0-9](0)?)\.pndsn\.com$/;
+
+      assert(regex.test(origin), `"${origin}" doesn't match the regex ${regex}`);
+    });
+  });
+
+  describe('if useDomainSharding is set to false', () => {
+    it('shouldn\'t generate sharded domains based on custom origin', () => {
+      const config = new Config({ setup: { ssl: true, origin: 'example.pubnubapi.com', useDomainSharding: false } })
+      const networking = new Networking({})
+      networking.init(config);
+
+      const origin = networking.getStandardOrigin();
+
+      assert.equal(origin, 'https://example.pubnubapi.com');
+    });
+
+    it('shouldn\'t generate sharded domains based on default origin', () => {
+      const config = new Config({ setup: { ssl: true, useDomainSharding: false } })
+      const networking = new Networking({})
+      networking.init(config);
+
+      const origin = networking.getStandardOrigin();
+
+      assert.equal(origin, 'https://ps.pndsn.com');
+    });
+  });
+
+  describe('if useDomainSharding is set to true', () => {
+    it('should generate sharded domains based on custom origin', () => {
+      const config = new Config({ setup: { ssl: true, origin: 'example.pubnubapi.com', useDomainSharding: true } })
+      const networking = new Networking({})
+      networking.init(config);
+
+      const origin = networking.getStandardOrigin();
+
+      const regex = /^https:\/\/example\-([0-9](0)?)\.pubnubapi\.com$/;
+
+      assert(regex.test(origin), `"${origin}" doesn't match the regex ${regex}`);
+    });
+
+    it('should generate sharded domains based on default origin', () => {
+      const config = new Config({ setup: { ssl: true, useDomainSharding: true } })
+      const networking = new Networking({})
+      networking.init(config);
+
+      const origin = networking.getStandardOrigin();
+
+      const regex = /^https:\/\/ps([0-9](0)?)\.pndsn\.com$/;
+
+      assert(regex.test(origin), `"${origin}" doesn't match the regex ${regex}`);
+    });
+  })
+
+});
+
+
 describe('keep-alive agent', () => {
   before(() => nock.disableNetConnect());
   after(() => nock.enableNetConnect());
 
   const setupNetwork = (shouldSecure, enableKeepAlive) => {
-    const config = { origin: 'ps.pndsn.com', secure: shouldSecure, keepAlive: enableKeepAlive, logVerbosity: false };
+    const config = new Config({ setup: { origin: 'ps.pndsn.com', ssl: shouldSecure, keepAlive: enableKeepAlive, logVerbosity: false } });
     const networking = new Networking({ keepAlive, del, get, post, proxy });
     networking.init(config);
 
```

Comment:
Tests look great. I did think of one more test that may help which would validate the actual generated sub domains as 1-10 or 0-9 for internal vs external domains respectively by calling nextOrigin() repeatedly and also make sure when it gets to the max it returns to the min of 0 or 1.